pkg/discovery/nodes/nodes_test.go

```diff
@@ -4,29 +4,37 @@ import (
 	"testing"
 
 	"github.com/docker/docker/pkg/discovery"
-	"github.com/stretchr/testify/assert"
+
+	. "github.com/go-check/check"
 )
 
-func TestInitialize(t *testing.T) {
+// Hook up gocheck into the "go test" runner.
+func Test(t *testing.T) { TestingT(t) }
+
+type DiscoverySuite struct{}
+
+var _ = Suite(&DiscoverySuite{})
+
+func (s *DiscoverySuite) TestInitialize(c *C) {
 	d := &Discovery{}
```

Comment:
If you do always the same thing (here `d := &Discovery{}`), you can do something like :

```go
type DiscoverySuite struct{
    d *Discovery
}

func (s *DiscoverySuite) SetUpTest(c *check.C) {
    s.d = &Discovery{}
}
```